rmw_iceoryx_cpp/src/internal/iceoryx_name_conversion.cpp

```diff
@@ -101,12 +77,45 @@ get_name_n_type_from_iceoryx_service_description(
     return std::make_tuple(instance, service);
   } else {
     // ARA Naming
-    std::string delimiter_msg = "_ara_msgs/msg/";
+    std::string service_lowercase = service;
+    std::transform(
+      service_lowercase.begin(), service_lowercase.end(),
+      service_lowercase.begin(), ::tolower);
 
     return std::make_tuple(
       "/" + instance + "/" + service + "/" + event,
-      service + delimiter_msg + event);
+      service_lowercase + DELIMITER_MSG + event);
+  }
+}
+
+std::tuple<std::string, std::string, std::string> get_service_description_elements(
+  const std::string & topic_name,
+  const std::string & type_name)
+{
+  auto pos_delimiter_msg = type_name.find(DELIMITER_MSG);
+
+  // ROS 2.0 Naming
+  if (pos_delimiter_msg == std::string::npos) {
+    // service, instance, event
+    return std::make_tuple(type_name, topic_name, "data");
+  }
+
+  // Could check more detailed!!
+  auto service_lowercase = type_name.substr(0, pos_delimiter_msg);
```

Comment:
This is still confusing to me. Why is `service_lowercase` and `service` calculated differently?

`type_name.substr(0, pos_delimiter_msg)` vs `topic_name.substr(pos_package_name, service_lowercase.length())`

Reply:
The service name is present two times. One in the type name and one in the topic name. In the type name  we know that its length is till the delimiter. But we do not know capital and small letters. In the topic name we have correct capital and lowercase letters but we do not know the length.